r-package/grf/vignettes/diagnostics.Rmd

```diff
@@ -31,22 +33,73 @@ Y <- pmax(X[, 1], 0) * W + X[, 2] + pmin(X[, 3], 0) + rnorm(n)
 cf <- causal_forest(X, Y, W)
 ```
 
-The overlap assumptions requires a positive probability of treatment for each $X_i$. We should not be able to deterministically decide the treatment status of an individual based on its covariates, meaning none of the estimated propensity scores should be close to one or zero. One can check this with a histogram:
+The overlap assumption requires a positive probability of treatment for each $X_i$. We should not be able to deterministically decide the treatment status of an individual based on its covariates, meaning none of the estimated propensity scores should be close to one or zero. One can check this with a histogram:
 
 ```{r}
 hist(e.hat <- cf$W.hat)
 ```
 
-One can also check that the covariates are balanced across the control and treated group by plotting the histogram $X_i | W_i = k$, overlaid here for each feature:
+One can also check that the covariates are balanced across the treated and control group by plotting the histograms of the inverse-propensity weighted samples $\frac{X_i | W_i = 1}{\hat e_i}$ and $\frac{X_i | W_i = 0}{1 - \hat e_i}$, overlaid here for each feature:
 
 ```{r, fig.height = 25}
-par(mar = c(5, 5, 1, 1), oma = c(0, 0, 2, 0), mfrow = c(p/2, 2))
+par(mar = c(5, 5, 1, 1), oma = c(0, 0, 2, 0), mfrow = c(p / 2, 2))
+ipw.control <- 1 / (1 - e.hat[W == 0])
+ipw.treated <- 1 / e.hat[W == 1]
 plots <- lapply(1:p, function (i) {
-  control.hist <- hist(X[W == 0, i], plot = FALSE)
-  treated.hist <- hist(X[W == 1, i], plot = FALSE)
+  control.hist <- hist(X[W == 0, i] * ipw.control, plot = FALSE)
```

Comment:
This doesn't give one a weighted histogram. It looks like the basic `hist` command in R can't do weighted histograms, and we need to use ggplot (e.g., https://stackoverflow.com/a/19843767/2292304). There are other packages that offer a weighted histograms, but I'd rather depend on ggplot than on some packages I've never heard of before.